### DISCUSSION TITLE
Allow entry selection in URL field

### DISCUSSION THREAD

**Discussion by kruncher** [craftcms/cms#11637] [Ideas] [answered]
Whenever allowing users to select links we find ourselves needing 2 fields; "Link URL" or "Target entry" with a comment telling the content editor to provide just one of these.

Would it be possible to allow for the selection of entries and then to have the link work in the same way that they do with Redactor fields? This would make it a lot easier to define link fields in all kinds of places; especially navigation.

**Comment by alexanderloewe** [comment]
You can use this plugin to do that:
https://plugins.craftcms.com/typedlinkfield

I have pushed the idea to the Craft CMS team to make this plugin a part of the core setup, what I would love to see.

**Comment by cherrykoda** [comment]
Attempting to combine requests to the bigger thread: https://github.com/craftcms/cms/discussions/9062

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.3.0-beta.1 is out with a new Link field type, which addresses this feature request 🎉 (#15251).

We’re targeting August 6 for the GA release. If you’d like to test ahead of time, change your `craftcms/cms` requirement in composer.json to:

```json
"craftcms/cms": "^5.3.0-beta.1",
```

and then run `composer update`.

**Reply by kruncher** [reply]
This looks fantastic thank you :)

**Reply by alexanderbuergin** [reply]
Thanks.
